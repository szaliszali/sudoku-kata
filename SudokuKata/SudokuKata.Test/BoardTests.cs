using System;
using ApprovalTests;
using NUnit.Framework;

namespace SudokuKata.Test;

[TestFixture]
internal class BoardTests
{
    [Test]
    public void EmptyBoardToString() => Approvals.Verify(new CharArrayBoard());

    [Test]
    public void EmptyBoardCode() => Approvals.Verify(new CharArrayBoard().Code);

    [Test]
    public void RandomBoard()
    {
        var rng = new Random(1);
        var sut = new RandomBoard(rng);
        Approvals.Verify((sut, sut.Code));
    }

    [Test]
    public void SetCellToDigit()
    {
        var sut = new CharArrayBoard();
        sut.Set(1, 2, 3);
        Approvals.Verify(sut);
    }

    [Test]
    public void ClearCell()
    {
        var sut = new CharArrayBoard();
        sut.Set(1, 2, 3);
        sut.Set(1, 2, 0);
        Approvals.Verify(sut);
    }

    [Test]
    public void ConstructFromStateArray()
    {
        var state = new[] { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0 };
        var sut = new CharArrayBoard(state);
        Approvals.Verify((sut, sut.Code));
    }

    [Test]
    public void ConstructFromStateArray_StateIsPreserved()
    {
        var state = new[] { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0 };
        var sut = new CharArrayBoard(state);
        Assert.That(sut.State, Is.EqualTo(state));
    }
}
