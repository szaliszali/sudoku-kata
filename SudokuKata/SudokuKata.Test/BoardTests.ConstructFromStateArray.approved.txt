(+---+---+---+
|.12|345|678|
|9.1|234|567|
|89.|123|456|
+---+---+---+
|789|.12|345|
|678|9.1|234|
|567|89.|123|
+---+---+---+
|456|789|.12|
|345|678|9.1|
|234|567|89.|
+---+---+---+, 012345678901234567890123456789012345678901234567890123456789012345678901234567890)